internal/functions/deploy/deploy.go

```diff
@@ -92,20 +84,37 @@ func GetFunctionConfig(slugs []string, importMapPath string, noVerifyJWT *bool,
 		if len(importMapPath) > 0 {
 			function.ImportMap = importMapPath
 		} else if len(function.ImportMap) == 0 {
-			denoJsonPath := filepath.Join(functionDir, "deno.json")
-			denoJsoncPath := filepath.Join(functionDir, "deno.jsonc")
-			if _, err := fsys.Stat(denoJsonPath); err == nil {
-				function.ImportMap = denoJsonPath
-			} else if _, err := fsys.Stat(denoJsoncPath); err == nil {
-				function.ImportMap = denoJsoncPath
-			} else if fallbackExists {
-				function.ImportMap = utils.FallbackImportMapPath
+			if dedicatedFunctionPath, err := utils.GetImportsFilePath(functionDir, fsys); err == nil {
+				function.ImportMap = dedicatedFunctionPath
+			} else if fallbackFunctionPath, err := utils.GetImportsFilePath(utils.FunctionsDir, fsys); err == nil {
+				function.ImportMap = fallbackFunctionPath
 			}
 		}
 		if noVerifyJWT != nil {
 			function.VerifyJWT = cast.Ptr(!*noVerifyJWT)
 		}
 		functionConfig[name] = function
 	}
+	// Check validity of ImportMap paths
+	functionsWithFallback := []string{}
+	if fallbacksPath, err := utils.GetImportsFilePath(utils.FunctionsDir, fsys); err == nil {
```

Comment:
If we really want to support fallback import map for deno.json, then this should be moved to pkg/config.

So each function config's import_map field will point to the fully resolved path after considering all possible fallbacks.

Reply:
**question**

It seems like we didn't supported this fallback before ? I don't know how that was handled with branching 🤔 

I have two question about this:

1. Won't it have side effects with the diffing of the config in such case ?
2. I think we can't import utils within pkg/config without causing a cycling deps issue. What would be the best approach here ?